sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,264 @@
+import { Injectable } from "@nestjs/common";
+import {
+  CASService,
+  PendingInvoicePayload,
+  InvoiceLineDetail,
+  SendPendingInvoicesResponse,
+} from "@sims/integrations/cas";
+import { CAS_BAD_REQUEST } from "@sims/integrations/constants";
+import {
+  CASInvoice,
+  CASInvoiceBatchApprovalStatus,
+  CASInvoiceStatus,
+  User,
+} from "@sims/sims-db";
+import {
+  processInParallel,
+  CustomNamedError,
+  getAbbreviatedDateOnlyFormat,
+} from "@sims/utilities";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { DataSource, Repository } from "typeorm";
+import { InjectRepository } from "@nestjs/typeorm";
+import { SystemUsersService } from "@sims/services";
+
+@Injectable()
+export class CASInvoiceService {
+  constructor(
+    private readonly systemUsersService: SystemUsersService,
+    private readonly casService: CASService,
+    @InjectRepository(CASInvoice)
+    private readonly casInvoiceRepo: Repository<CASInvoice>,
+    private readonly dataSource: DataSource,
+  ) {}
+  /**
+   * Get the list of pending invoices to be sent from the approved batch.
+   * @returns list of pending invoices.
+   */
+  async getPendingInvoices(): Promise<CASInvoice[]> {
+    const pendingInvoices = await this.dataSource
+      .getRepository(CASInvoice)
+      .find({
+        select: {
+          id: true,
+          casSupplier: {
+            id: true,
+            supplierNumber: true,
+            supplierAddress: { supplierSiteCode: true },
+          },
+          createdAt: true,
+          invoiceNumber: true,
+          disbursementReceipt: {
+            id: true,
+            fileDate: true,
+            createdAt: true,
+          },
+          casInvoiceBatch: {
+            id: true,
+            batchName: true,
+            batchDate: true,
+            createdAt: true,
+          },
+          casInvoiceDetails: {
+            id: true,
+            valueAmount: true,
+            casDistributionAccount: {
+              id: true,
+              operationCode: true,
+              awardValueCode: true,
+              distributionAccount: true,
+            },
+          },
+        },
+        relations: {
+          casInvoiceBatch: true,
+          disbursementReceipt: true,
+          casSupplier: true,
+          casInvoiceDetails: {
+            casDistributionAccount: true,
+          },
+        },
+        where: {
+          casInvoiceBatch: {
+            approvalStatus: CASInvoiceBatchApprovalStatus.Approved,
+          },
+          invoiceStatus: CASInvoiceStatus.Pending,
+        },
+      });
+    return pendingInvoices;
```

Comment:
The limit is missing from this query, and since we will be adding a limit, we should also add an order by. The order can be FIFO, which means invoice creation date ASC, does it make sense?